src/kafka/decam.rs

```diff
@@ -0,0 +1,247 @@
+use crate::{
+    conf,
+    kafka::base::{consume_partitions, AlertConsumer, AlertProducer, ConsumerError},
+    utils::{
+        data::count_files_in_dir,
+        o11y::{as_error, log_error},
+    },
+};
+use indicatif::ProgressBar;
+use rdkafka::config::ClientConfig;
+use rdkafka::producer::{FutureProducer, FutureRecord, Producer};
+use redis::AsyncCommands;
+use tracing::{error, info, instrument};
+
+const DECAM_SERVER_URL: &str = "localhost:9092";
+
+pub struct DecamAlertConsumer {
+    output_queue: String,
+    n_threads: usize,
+    max_in_queue: usize,
+    group_id: String,
+    server: String,
+    config_path: String,
+}
+
+impl DecamAlertConsumer {
+    pub fn new(
+        n_threads: usize,
+        max_in_queue: Option<usize>,
+        output_queue: Option<&str>,
+        group_id: Option<&str>,
+        server: Option<&str>,
+        config_path: &str,
+    ) -> Self {
+        if 15 % n_threads != 0 {
+            panic!("Number of threads should be a factor of 15");
+        }
+        let max_in_queue = max_in_queue.unwrap_or(15000);
+
+        let output_queue = output_queue
+            .unwrap_or("DECAM_alerts_packets_queue")
+            .to_string();
+        let mut group_id = group_id.unwrap_or("example-ck").to_string();
+        let server = server.unwrap_or(DECAM_SERVER_URL).to_string();
+
+        group_id = format!("{}-{}", "decam", group_id);
+
+        info!(
+            "Creating AlertConsumer with {} threads, output_queue: {}, group_id: {}, server: {}",
+            n_threads, output_queue, group_id, server
+        );
+
+        DecamAlertConsumer {
+            output_queue,
+            n_threads,
+            max_in_queue,
+            group_id,
+            server,
+            config_path: config_path.to_string(),
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl AlertConsumer for DecamAlertConsumer {
+    fn default(config_path: &str) -> Self {
+        Self::new(1, None, None, None, None, config_path)
+    }
+
+    #[instrument(skip(self))]
+    async fn consume(&self, timestamp: i64) {
+        let partitions_per_thread = 15 / self.n_threads;
+        let mut partitions = vec![vec![]; self.n_threads];
+        for i in 0..15 {
+            partitions[i / partitions_per_thread].push(i as i32);
+        }
+
+        // DECAM uses nightly topics, and no user/pass (IP whitelisting)
```

Comment:
For my own understanding: this is exactly the same as for ZTF, then, right?

Reply:
Truth is right now they dont even have topic whatsoever, so Ill just replace that comment